broadinstitute/firecloud-orchestration
# GAWB-2767 Index consent codes as a structured object

Addresses https://broadinstitute.atlassian.net/browse/GAWB-2767

## Changes
* This PR adds one more element to the workspace document that is indexed to ES. In the case where there are consent code library attributes, we collect the valid ones and flesh that out into a full object. 

## Testing
* Run this against a custom index if running locally.
* Make sure there is at least one workspace that is catalogued with answers to the data use questionnaire.
* Run a reindex
* Search your local index for something like this:
```
http://elasticsearch5a3.dsde-dev.broadinstitute.org:9200/librarylocalgr/dataset/_search?q=_id=df7d1e4a-0956-436c-bf29-5d17a6af9483
```
You should see an indexed workspace with the following object:
```
          "library:structuredUseRestriction": {
            "DS": [
              "Diabetes Mellitus"
            ], 
            "GRU": false, 
            "HMB": true, 
            "NAGR": false, 
            "NCTRL": false, 
            "NCU": false, 
            "NDMS": false, 
            "NPU": false, 
            "RS-FM": false, 
            "RS-G": false, 
            "RS-M": false, 
            "RS-PD": false, 
            "RS-POP": [
              "N/A"
            ]
          }, 
``` 

---
Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [x] I've followed [the instructions](https://github.com/broadinstitute/firecloud-orchestration/blob/develop/CONTRIBUTING.md#api-changes) if I've made any changes to the API, _especially_ if they're breaking changes
- [x] I've updated the RC_XXX release ticket with any manual steps required to release this change
- [ ] I've updated the [FISMA documentation](https://github.com/broadinstitute/firecloud-orchestration/blob/develop/CONTRIBUTING.md#fisma-documentation-changes) if I've made any security-related changes, including auth, encryption, or auditing

In all cases:

- [x] Squash and merge; you can delete your branch after this **unless it's for a hotfix**. In that case, don't delete it!
